dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -65,6 +72,115 @@ window.Dashboard = {
 		Marbles.history.navigate('login'+ redirectPath);
 	},
 
+	__catchInsecurePingResponse: function(httpsArgs, resolve) {
+		var httpsXhr = httpsArgs[1], self = this,
+			handleSuccess, handleError, p;
+
+		handleSuccess = function (httpArgs, resolve) {
+			var httpXhr = httpArgs[1];
+			// https did not work but http did...something is wrong with the cert
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+			resolve();
+		};
+		handleError = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			if (httpXhr.status === 0) {
+				// https is failing as well...service is unavailable
+				self.Dispatcher.handleAppEvent({
+					name: "SERVICE_UNAVAILABLE",
+					status: httpXhr.status
+				});
+			}
+			// https did not work but http did without a network error
+			// => missing ssl exception for controller
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+			resolve();
+		};
+
+		if (httpsXhr.status === 0) {
+			// https is unavailable, let's see if http works
+			self.client.ping("controller", "http").catch(function(args) {
+				handleError(args, resolve);
+			}).then(function(args){
+				handleSuccess(args, resolve)
+			});
+			return;
```

Comment:
Move this return in front of `self.client.ping` so the resulting promise is returned.